ribs/archives/_cvt_archive.py

```diff
@@ -164,16 +166,55 @@ def __init__(self,
         self._chunk_size = chunk_size
 
         if custom_centroids is None:
-            if not isinstance(samples, int):
-                # Validate shape of custom samples. These are ignored when
-                # `custom_centroids` is provided.
-                samples = np.asarray(samples, dtype=self.dtype)
-                if samples.shape[1] != self._measure_dim:
-                    raise ValueError(
-                        f"Samples has shape {samples.shape} but must be of "
-                        f"shape (n_samples, len(ranges)={self._measure_dim})")
-            self._samples = samples
-            self._centroids = None
+            if centroid_method == "kmeans":
+                if not isinstance(samples, int):
+                    # Validate shape of custom samples. These are ignored when
```

Comment:
I think we can take out "These are..." Instead, in the docstring above, mention that samples is only used when `centroid_method="kmeans"`